src/military/military.cpp

```diff
@@ -3961,46 +3968,24 @@ void army_arrives_in_province(sys::state& state, dcon::army_id a, dcon::province
 		}
 
 		// start battle
-		dcon::land_battle_id gather_to_battle;
-		dcon::war_id battle_in_war;
-
+		auto gather_to_battle = dcon::land_battle_id{};
+		auto battle_in_war = dcon::war_id{};
 		for(auto o : state.world.province_get_army_location(p)) {
 			if(o.get_army() == a)
 				continue;
-			if(o.get_army().get_is_retreating() || o.get_army().get_black_flag() || o.get_army().get_navy_from_army_transport() || o.get_army().get_battle_from_army_battle_participation())
+			if(!army_can_join_land_battle(state, o.get_army()))
 				continue;
 
 			auto other_nation = o.get_army().get_controller_from_army_control();
 
-			if(bool(owner_nation) != bool(other_nation)) { // battle vs. rebels
```

Comment:
This logic you have removed was not redundant. It works slightly differently for rebels